packages/next-app/src/contexts/PoolMetadataContext.tsx

```diff
@@ -50,17 +27,18 @@ interface PoolMetadataContextType {
   selectedPool: string | null;
   handleSetPool: (poolId: string) => void;
   isKeyUnique: (key: string) => boolean;
+  handleSetMetadata: (data: PoolMetadataAttribute[]) => void;
 }
 
 export const PoolMetadataContext = createContext({} as PoolMetadataContextType);
 
 export function PoolMetadataProvider({ children }: { children: ReactNode }) {
   const [selectedPool, setSelectedPool] = useState<string>("");
-  // const { data: poolMetadata } = metadataGql.useMetadataPool({
-  //   poolId: selectedPool,
-  // });
-  // cellData = getDataFromIpfs(poolMetadata.CID)
-  const [metadata, setMetadata] = useState<PoolMetadataAttribute[]>(cellData);
+  const [metadata, setMetadata] = useState<PoolMetadataAttribute[]>([]);
+
+  function handleSetMetadata(data: PoolMetadataAttribute[]) {
+    setMetadata(data);
+  }
```

Comment:
useless function definition?

Reply:
From what I know it's generally considered a good practice to pass a function that handles setState through a context, rather than passing the setState function itself

Reply:
oh really, I didn't know that! Curious about the reasoning behind it. Do you have any links you could point me out to?